Main.java

```diff
@@ -38,38 +38,43 @@ private static double readIn(Scanner sc, String msg) {
     }
 
     private static void battle(Unit unit1, Unit unit2) {
-        double as1 = unit1.getAs();
-        double as2 = unit2.getAs();
         String name1 = unit1.getName();
         String name2 = unit2.getName();
-        boolean bothAlive = true;
-
-        // Amíg mindkettő él
-        int i = -1;
+        boolean bothAlive = true; // mindkettő él
+        boolean isFirstRound = true; // első kör
+        double defaultAs = unit1.getAs();
+        double defaultAs2 = unit2.getAs();
         while (bothAlive) {
-            i++;
-            // Egyszerre megütik egymást 1. ütés!
-            if (i <= 0) {
+            // Ha első kör akkor egyszerre megütik egymást 1. ütés!
+            if (isFirstRound) {
+                isFirstRound = false;
                 unit1.attack(unit2);
                 unit2.attack(unit1);
                 System.out.println("\nA Csata elkezdődött! " + name1 + " és " + name2 + " megtámadták egymást. Életük: "
-                        + name1 + " " + unit1.getHP() + ", " + name2 + " " + unit2.getHP());
+                        + name1 + " " + unit1.getHp() + ", " + name2 + " " + unit2.getHp());
             }
-            // Ha túlélték akkor ütnek amikor lejár az Attackspeed pl.: attackspeed = 1,
-            // akkor minden i-nél üthetünk (1i = 1mp)
-            if (i > 0 && i % as1 == 0) {
+            // Attack speed számolása, melyiké kisebb --> az üthet elősször
+            double lowestAs = Math.min(unit1.getAs(), unit2.getAs());
+            unit1.setAs(unit1.getAs() - lowestAs);
+            unit2.setAs(unit2.getAs() - lowestAs);
+            // Ütések
+            if (unit1.getAs() == 0) {
                 unit1.attack(unit2);
-                System.out.println(name1 + " megtámadta " + name2 + ", így " + name2 + " élete - " + unit2.getHP());
+                unit1.setAs(defaultAs);
+                System.out.println(name1 + " megtámadta " + name2 + ", így " + name2 + " élete - " + unit2.getHp());
             }
-            if (i > 0 && i % as2 == 0) {
+            if (unit2.getAs() == 0) {
                 unit2.attack(unit1);
-                System.out.println(name2 + " megtámadta " + name1 + ", így " + name1 + " élete - " + unit1.getHP());
+                unit2.setAs(defaultAs2);
```

Comment:
u.a.

Reply:
Ez a setAs nem jó értékre állítja az attack speed-et. Úgy is jó, ha csak az attack függvényből törlöm és battle marad így? @hegyhati

Reply:
igen, most latom, hogy az sajat magara allitotta be. Igazabol van egy curent, meg egy default AS, de jo lesz akkor igy.